test/state/host.cpp

```diff
@@ -256,7 +256,7 @@ std::optional<evmc_message> Host::prepare_message(evmc_message msg)
                     if (holds_alternative<EOFValidationError>(header_or_error))
                         return {};  // Light early exception.
 
-                    const auto& header = std::get<EOF1Header>(header_or_error);
+                    const auto header = std::get<EOF1Header>(header_or_error);
```

Comment:
So `std::get` is actually throwing, I guess that's why the warning. 

1. The reference shouldn't affect this I think, so would be good to report to clang-tidy if it's possible to reproduce on some minimal example
2. To work around we can use `get_if`
```cpp
                   const auto* header = std::get_ig<EOF1Header>(&header_or_error);
                    if (!header)
                        return {};  // Light early exception.
```
3. I think it makes sense to add `noexcept` to `prepare_message()`

Reply:
But in `validate_eof1` a similar pattern is OK, which is what throws me off completely. Thanks for the suggestions, I'll poke around this